src/components/data-pane/field-list.tsx

```diff
@@ -58,6 +71,22 @@ class FieldListBase extends React.PureComponent<FieldListProps, {}> {
       payload: {fieldDef: fieldDef}
     });
   }
+
+  private getFilter(fieldDef: ShelfFieldDef, domain: any[]) {
```

Comment:
This seems like a controller logic, which better belongs in reducer?

Why? -- Having logic in the reducer means that we can easily test the logic in unit test independent from the UI.